src/app/modules/sgid/sgid.controller.ts

```diff
@@ -46,6 +46,32 @@ export const handleLogin: ControllerHandler<
   }
 
   const form = formResult.value
+  // @TODO move this to separate route.
+  if (form.authType === FormAuthType.SGID_MyInfo) {
+    const jwtResult = await SgidService.retrieveAccessToken(code).andThen(
+      (data) => SgidService.createSgidMyInfoJwt(data.accessToken, rememberMe),
+    )
+
+    if (jwtResult.isErr()) {
+      logger.error({
+        message: 'Error while handling login via sgID (MyInfo)',
+        meta,
+        error: jwtResult.error,
+      })
+      res.cookie('isLoginError', true)
+      return res.redirect(target)
+    }
+
+    const { maxAge, jwt } = jwtResult.value
+    res.cookie(SGID_MYINFO_COOKIE_NAME, jwt, {
+      maxAge,
+      httpOnly: true,
+      sameSite: 'lax', // Setting to 'strict' prevents Singpass login on Safari, Firefox
+      secure: !config.isDev,
+      ...SgidService.getCookieSettings(),
+    })
+    return res.redirect(target)
+  }
   if (form.authType !== FormAuthType.SGID) {
```

Comment:
Probably should shift this before L50(`if (form.authType === FormAuthType.SGID_MyInfo) {`), so that if form type is neither SGID or SGID_MyInfo, it is logged appropriately and `isLoginError` set

The handling for the two cases `FormAuthType.SGID_MyInfo` and `FormAuthType.SGID` then follows thereafter